### PR TITLE
Simpler & safer quote usage

### PR DESCRIPTION
**Pull request by jglick**
Untested.

### CODE REVIEW

File: Jenkinsfile.release [hunk 1]
```diff
@@ -25,7 +25,7 @@ pipeline {
       kubernetes {
         label 'jenkins-release'
         defaultContainer 'jnlp'
-        yaml """
+        yaml '''
```
Comment (jglick) [inline Jenkinsfile.release]: Always use single quotes unless you specifically intend to do Groovy interpolation (`GString`s).

File: Jenkinsfile.release [hunk 2]
```diff
@@ -72,7 +72,7 @@ spec:
       REPOSITORY_USERNAME           = credentials('maven-repository-username')
       REPOSITORY_PASSWORD           = credentials('maven-repository-password')
       SIGN_ALIAS                    = 'jenkins'
-      SIGN_KEYSTORE                 = "${ env.WORKSPACE + '/jenkins.pfx' }"
+      SIGN_KEYSTORE                 = "$WORKSPACE/jenkins.pfx"
```
Comment (jglick) [inline Jenkinsfile.release]: Keep it simple!

File: Jenkinsfile.release [hunk 3]
```diff
@@ -96,15 +96,19 @@ spec:
 
         steps {
           container('azure-cli') {
-            sh "az login --service-principal \
-              -u '${ env.AZURE_VAULT_CLIENT_ID }' \
-              -p '${ env.AZURE_VAULT_CLIENT_SECRET }' \
-              -t '${ env.AZURE_VAULT_TENANT_ID }'"
+            sh '''
```
Comment (jglick) [inline Jenkinsfile.release]: Avoid passing interpolated strings to `sh`. [Discussion](https://github.com/jenkinsci/credentials-binding-plugin/blob/22fcd37c62bd880bcacec1b59065ec222aee67d5/src/main/resources/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep/help.html#L28-L42)

File: Jenkinsfile.release [hunk 4]
```diff
@@ -183,20 +189,24 @@ spec:
...
           container('azure-cli') {
-            sh "gpg --verify '${ env.JENKINS_ASC }' '${ env.JENKINS_WAR }'"
-            sh "unzip -qc '${ env.JENKINS_WAR }' META-INF/MANIFEST.MF | grep 'Jenkins-Version' | awk \'{print \$2}\'"
-            sh "az storage container upload \
-              --acount-name '${ env.AZURE_STORAGE_ACCOUNT }' \
-              --share-name '${ env.AZURE_SHARE_NAME }' \
-              --path '${ env.VERSION/jenkins.war }' \
+            sh 'gpg --verify "$JENKINS_ASC" "$JENKINS_WAR"'
+            sh '''
+              unzip -qc "$JENKINS_WAR" META-INF/MANIFEST.MF | grep 'Jenkins-Version' | awk '{print $2}'
```
Comment (jglick) [inline Jenkinsfile.release]: More readable I think.

I am keeping _shell_ quotes around anything that might include the workspace path, for the unlikely case that the agent FS root includes a space in its name. If you are running in K8S with a predictable environment you could probably drop them for readability, though.

File: Jenkinsfile.release [hunk 5]
```diff
@@ -133,9 +138,11 @@ spec:
...
               // Maven Release requires gpg key with password password and a certificate key with password
-              sh 'scripts/buildJenkins.bash --configureGPG'
-              sh 'scripts/buildJenkins.bash --configureGit'
-              sh 'scripts/buildJenkins.bash --prepareRelease'
+              sh '''
+                scripts/buildJenkins.bash --configureGPG
+                scripts/buildJenkins.bash --configureGit
+                scripts/buildJenkins.bash --prepareRelease
+              '''
               script {
```
Comment (jglick) [inline Jenkinsfile.release]: BTW you could get rid of the `script` block by externalizing this processing: the script doing the `git push` could just load `scm.tag` from `release.properties` itself.

File: Jenkinsfile.release (hunk 3 above)
Comment (olblak) [reply Jenkinsfile.release]: Good to know thanks

File: Jenkinsfile.release (hunk 2 above)
Comment (olblak) [reply Jenkinsfile.release]: :+1:

File: Jenkinsfile.release (hunk 1 above)
Comment (olblak) [reply Jenkinsfile.release]: ack

File: Jenkinsfile.release (hunk 5 above)
Comment (olblak) [reply Jenkinsfile.release]: Do you mean by letting maven release to handle it from pom.xml?
```
  <scm>
    <connection>scm:git:git://github.com/jenkinsci/jenkins.git</connection>
    <developerConnection>scm:git:ssh://git@github.com/jenkinsci/jenkins.git</developerConnection>
    <url>https://github.com/jenkinsci/jenkins</url>
    <tag>${scmTag}</tag>
  </scm>
```

Comment (olblak) [reply Jenkinsfile.release]: I am not sure to understand your point here

Comment (jglick) [reply Jenkinsfile.release]: I mean that if you deleted this `script` block, the corresponding shell script fragment

```sh
git push origin $RELEASE_SCM_TAG
```

could do the equivalent computation on its own by parsing `release.properties`.

**Pull request merged**